packages/hierarchy-builder/src/test/HierarchyProvider.test.ts

```diff
@@ -147,28 +149,20 @@ describe("HierarchyProvider", () => {
       });
 
       const queryTimeout1 = new ResolvablePromise();
-      queryExecutor.createQueryReader.onFirstCall().returns({
-        async *[Symbol.asyncIterator](): AsyncIterableIterator<ECSqlQueryRow> {
-          await queryTimeout1;
-        },
-      });
+      queryExecutor.createQueryReader.onFirstCall().returns(eachValueFrom<ECSqlQueryRow>(from(queryTimeout1).pipe(mergeMap(() => EMPTY))));
```

Comment:
I think it should be possible to replace `from(queryTimeout1).pipe(mergeMap(() => EMPTY)))` with Subject<ECSqlQueryRow>. Maybe it would make this test a little bit easier to understand?

Reply:
made a change to stop using `rxjs` for this altogether